docker-compose.yml

```diff
@@ -0,0 +1,14 @@
+version: '3.7'
+services:
+  citybase-flask-app:
+    #command: ["tail", "-f", "/dev/null"]
+    command: ["/usr/bin/python3", "/root/app/app.py"]
+    container_name: citybase-flask-app
+    build: .
+    restart: unless-stopped
```

Comment:
My understanding is that `unless-stopped` is the functional same as `always` except it allows for a developer to bring down the service with `docker compose down` and not have it immediately restart. `unless-stopped` will start back up on the docker service restarting. Manually stopped is maybe not the best name they picked, but I believe it means "expressly stopped by a human."

Reply:
this has been a mystery to me for some time, so i just ran a few tests and found [this helpful article on the behavior](https://www.cloudbees.com/blog/ensuring-containers-are-always-running-with-dockers-restart-policy).

we use `always` in Moped's local stack. you can manually `stop` and `kill` the hasura container and it will not restart. if you stop the the docker daemon while the containers exist (running or no), they will spin back up when you restart the daemon. 

`unless-stopped` seems like it has almost the exact behavior, but the containers will not come back up if you manually stopped the containers and then restart the docker daemon.

i sleep better with restart `always` since it seems to have the highest likelihood of stuff coming back online after an unexpected reboot, but we are definitely splitting hairs 🤷

Reply:
That's a great article John, it spells it out and your testing does as well. Since we're going to be running this in production on an EC2 instance (unless we go for fargate which has AWS managed restart policies) - `restart: always` does sound good. 

As to your comment about moped's local stack; I've wondered why we have a restart policy at all. I'd love to not have to stop the services anytime I restart my computer because they were `restart: always`. I have a feeling that that restart policy is required to make `hasura-cluster` work correctly - but I can't remember the details of why I believe that. A mystery for another day.